File: frontend/fundrbolt-admin/src/types/ticketSalesImport.ts
/**
 * Types for ticket sales import feature
 */

export enum ImportRowStatus {
  CREATED = 'created',
  SKIPPED = 'skipped',
  ERROR = 'error',
}

export enum IssueSeverity {
  ERROR = 'error',
  WARNING = 'warning',
}

export interface PreflightIssue {
  row_number: number
  field_name?: string | null
  severity: IssueSeverity
  message: string
  raw_value?: string | null
}

export interface PreflightResult {
  preflight_id: string
  detected_format: string
  total_rows: number
  valid_rows: number
  error_rows: number
  warning_rows: number
  issues: PreflightIssue[]
  warnings: PreflightIssue[]
  error_report_url?: string | null
}

export interface ImportRowResult {
  row_number: number
  external_sale_id?: string | null
  purchaser_name?: string | null
  status: ImportRowStatus
  message: string
}

export interface ImportResult {
  batch_id: string
  created_rows: number
  skipped_rows: number
  failed_rows: number
  warnings: PreflightIssue[]
}

export interface TicketSaleImportRow {
  ticket_type: string
  purchaser_name: string
  purchaser_email: string
  quantity: number
  total_amount: number
  purchase_date: string
  external_sale_id: string
  purchaser_phone?: string | null
  fee_amount?: number | null
  payment_status?: string | null
  notes?: string | null
}

File: backend/app/schemas/ticket_sales_import.py
"""Schemas for ticket sales bulk import."""

from __future__ import annotations

from decimal import Decimal
from enum import Enum

from pydantic import BaseModel, Field


class ImportRowStatus(str, Enum):
    """Row-level import status."""

    CREATED = "created"
    SKIPPED = "skipped"
    ERROR = "error"


class IssueSeverity(str, Enum):
    """Severity level for validation issues."""

    ERROR = "error"
    WARNING = "warning"


class PreflightIssue(BaseModel):
    """Row-level validation issue from preflight."""

    row_number: int = Field(..., ge=1)
    field_name: str | None = None
    severity: IssueSeverity
    message: str
    raw_value: str | None = None


class PreflightResult(BaseModel):
    """Preflight validation results."""

    preflight_id: str
    detected_format: str
    total_rows: int = Field(..., ge=0)
    valid_rows: int = Field(..., ge=0)
    error_rows: int = Field(..., ge=0)
    warning_rows: int = Field(..., ge=0)
    issues: list[PreflightIssue] = Field(default_factory=list)
    warnings: list[PreflightIssue] = Field(default_factory=list)
    error_report_url: str | None = Field(
        None, description="Optional URL for downloadable error report"
    )


class ImportRowResult(BaseModel):
    """Result for a single imported row."""

    row_number: int = Field(..., ge=1)
    external_sale_id: str | None = None
    purchaser_name: str | None = None
    status: ImportRowStatus
    message: str


class ImportResult(BaseModel):
    """Import summary with created, skipped, and failed counts."""

    batch_id: str
    created_rows: int = Field(..., ge=0)
    skipped_rows: int = Field(..., ge=0)
    failed_rows: int = Field(..., ge=0)
    warnings: list[PreflightIssue] = Field(default_factory=list)


class TicketSaleImportRow(BaseModel):
    """Normalized import row parsed from the file."""

    # Required fields
    ticket_type: str = Field(..., min_length=1, max_length=200)
    purchaser_name: str = Field(..., min_length=1, max_length=200)
    purchaser_email: str = Field(..., min_length=1, max_length=200)
    quantity: int = Field(..., ge=1)
    total_amount: Decimal = Field(..., ge=0, decimal_places=2)
    purchase_date: str = Field(..., min_length=1)
    external_sale_id: str = Field(..., min_length=1, max_length=200)

    # Optional fields
    purchaser_phone: str | None = Field(None, max_length=50)
    fee_amount: Decimal | None = Field(None, ge=0, decimal_places=2)
    payment_status: str | None = Field(None, max_length=50)
    notes: str | None = Field(None, max_length=1000)
